network/wsNetwork.go

```diff
@@ -1762,3 +1775,14 @@ func SetUserAgentHeader(header http.Header) {
 	ua := fmt.Sprintf("algod/%d.%d (%s; commit=%s; %d) %s(%s)", version.Major, version.Minor, version.Channel, version.CommitHash, version.BuildNumber, runtime.GOOS, runtime.GOARCH)
 	header.Set(UserAgentHeader, ua)
 }
+
+// DoHTTP will check call to WaitAndAddConnectionTime to
+// make sure connectionsRateLimitingCount is not violated, and register the
+// connection time of the request, before sending the request to the server. 
+func DoHTTP(client *http.Client,
```

Comment:
Add this method as a method of WebsocketNetwork; i.e. 
```
func (wn *WebsocketNetwork) MakeHttpRequest(client *http.Client, request *http.Request )
```

which would extract the address from the request, take care of the phonebook waiting and updating and call the client.Do with the proper arguments.